### PR TITLE
Initial changes to allow video embed via wysiwyg with the new brightc…

### PR DESCRIPTION
**Pull request by sleepingmonk**
…ove API in brightcove 7.x-6.x module.  This has not been updated for playlists, and is just a starting point.

### CODE REVIEW

**Comment by q0rban** [comment]
Thanks so much for this, @sleepingmonk! Going to dive into peer review now. :)

File: brightcove-media-embed.tpl.php [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ * - $width
+ * - $height
+ *
+ * @see template_preprocess_brightcove_media_embed().
+ */
+?>
+<iframe
+  src="//players.brightcove.net/<?php print $account_id; ?>/<?php print $player_id; ?>_<?php print $embed; ?>/index.html?videoId=<?php print $video_id; ?>"
+  allowfullscreen webkitallowfullscreen mozallowfullscreen
+  width="<?php print $width; ?>" height="<?php print $height; ?>"></iframe>
```
Comment (q0rban) [inline brightcove-media-embed.tpl.php]: Missing EOL

File: brightcove_media.module [hunk 2]
```diff
@@ -119,9 +119,31 @@ function brightcove_media_theme($existing, $type, $theme, $path) {
...
+        'brightcove_id' => NULL,
+        'width' => BRIGHTCOVE_DEFAULT_VIDEO_WIDTH,
+        'height' => BRIGHTCOVE_DEFAULT_VIDEO_HEIGHT,
+        'embed' => 'default',
+      ],
+      'template' => 'brightcove-media-embed',
+    ],
+
   );
 }
```
Comment (q0rban) [inline brightcove_media.module:138]: Missing docblock.

File: brightcove_media.module [hunk 3]
```diff
@@ -564,27 +589,22 @@ function brightcove_media_list() {
...
       default:
-        $method = 'find_videos_by_text';
+        $method = '';
         $bc_find_params = array('text' => join(' ', explode(',', $params['filter']['keywords'])));
         break;
     }
-  }
-  else {
-    $method = 'find_all_videos';
+    if (isset($params['filter']['keywords'])) {
```
Comment (q0rban) [inline brightcove_media.module]: Should this be !empty()?

File: brightcove_media.module [hunk 4]
```diff
@@ -599,37 +619,41 @@ function brightcove_media_list() {
 
   if (!$result) {
     try {
-      $bc = brightcove_initialize();
-      $result = $bc->find($method, $bc_find_params);
+      $client = brightcove_client_load_or_default();
+      list($cms,,) = _brightcove_create_classes($client);
```
Comment (q0rban) [inline brightcove_media.module]: Do we need the extra commas in list if we're only grabbing the first one?

File: brightcove_media.module [hunk 5]
```diff
@@ -599,37 +619,41 @@ function brightcove_media_list() {
...
+      $result = $cms->listVideos($search, $sort, $limit, $offset);
```
Comment (q0rban) [inline brightcove_media.module]: Unless I'm reading this wrong, it looks like $search would be undefined unless `$params['filter']['keywords']` is set. And Where are $sort and $offset getting defined?

File: brightcove_media.module [hunk 6]
```diff
@@ -599,37 +619,41 @@ function brightcove_media_list() {
...
-  }
-  if (count($result)) {
-    foreach ($result as $video) {
-      $uri = file_stream_wrapper_uri_normalize("brightcove://{$video->id}");
-      $file = file_uri_to_object($uri);
-      $file->filename = $video->name;
-      $file->type = 'video';
-      if (!isset($file->fid)) {
-        $file->fid = 0;
+    if (count($result)) {
```
Comment (q0rban) [inline brightcove_media.module]: Let's use `if (!empty($result)) {`. It's more performant.

File: brightcove_media.module [hunk 7]
```diff
@@ -599,37 +619,41 @@ function brightcove_media_list() {
...
+
+        $file->preview = l($file->preview, 'media/browser', array(
+          'html' => TRUE,
+          'attributes' => array(
+            'data-uri' => $uri,
+          ),
+          'query' => array('render' => 'media-popup', 'uri' => $uri),
+        ));
+        $files[$uri] = $file;
+        $media[] = $file;
```
Comment (q0rban) [inline brightcove_media.module]: Can you document what's going on in here?

**Comment by q0rban** [comment]
I'm not seeing an update hook to move from 5.x to 6.x. The 6.x version of brightcove moves away from file entities to its own entity types, so we'd need to migrate existing to that, wouldn't we?

**Comment by sleepingmonk** [comment]
Thanks for the feedback.  I'll do the clean up bits and address your questions shortly.  This was part of a sprint, and I get more time on it this week.

**Comment by sleepingmonk** [comment]
Outdated

**Pull request closed without merging**